content/desktop/use-desktop/container.md

```diff
@@ -29,46 +29,92 @@ When you [inspect a container](#inspect-a-container), the **Stats** tab displays
+From the **Exec** tab, you can use the integrated terminal, on a running
+container, directly within Docker Desktop. You are able to quickly run commands
+within your container so you can understand its current state or debug when
+something goes wrong.
+
+Using the integrated terminal is the same as running one of the following commands:
+
+- `docker exec -it <container-id> /bin/sh`
+- `docker exec -it <container-id> cmd.exe` when accessing Windows containers
+- `docker debug <container-id>` when using debug mode
 
-Using the integrated terminal is the same as running `docker exec -it <container-id> /bin/sh`, or `docker exec -it <container-id> cmd.exe` if you are using Windows containers, in your external terminal. It also:
+The integrated terminal:
 
-- Automatically detects the default user for a running container from the image's Dockerfile. If no user is specified it defaults to `root`.
-- Persists your session if you navigate to another part of the Docker Dashboard and then return.
+- Persists your session and **Debug mode** setting if you navigate to another
+  part of the Docker Dashboard and then return.
 - Supports copy, paste, search, and clearing your session.
+- When not using debug mode, it automatically detects the default user for a
+  running container from the image's Dockerfile. If no user is specified, or
+  you're using debug mode, it defaults to `root`.
+
+#### Open the integrated terminal
 
 To open the integrated terminal, either:
-- Hover over your running container and select the **Show container actions** menu. From the drop-down menu, select **Open in terminal**.
+
+- Hover over your running container and select the **Show container actions**
+  menu. From the drop-down menu, select **Open in terminal**.
 - Select the container and then select the **Exec** tab.
 
-To use your external terminal, navigate to the **General** tab in **Settings** and select the **System default** option under **Choose your terminal**.
+To use your external terminal, navigate to the **General** tab in **Settings**
+and select the **System default** option under **Choose your terminal**.
+
+#### Open the integrated terminal in debug mode
+
+Debug mode is in [Beta](../../release-lifecycle.md#beta) and requires a [Pro, Team, or Business subcription](/subscription/details/). You must be signed in to use debug mode. Debug mode has several advantages, such as:
+
+- A customizable toolbox. The toolbox comes with many standard Linux tools
+  pre-installed, such as `vim`, `nano`, `htop`, and `curl`. For more details, see the [`docker debug` CLI reference](/reference/cli/docker/debug/).
+- The ability to access containers that don't have a shell, for example, slim or
+  distroless containers.
+
+To open the integrated terminal in debug mode, either:
+
+- Hover over your running container and select the **Show container actions**
```

Comment:
The same thing for Debug mode, if they click on the `Use Docker Debug`, in the menu actions, it means that the user is already logged in and a paid user.
User's shouldn't be able to see that as an option if they are not logged in and a paid user. The only way for them to get to the integrated terminal and use it in debug mode if they are not logged and or not a paid user would be to navigate to the exec tab, where they'd see a paywall banner and either be prompted to login. After logging in, if they are a paid user, they can enable Docker Debug mode. If they are not, they would then see an upgrade option where, if they do upgrade, they'd be able to enable Docker debug. 

I am soo sorry for such a long winded comment btw! I tried cutting it short but I tend to over explain. 🙇🏼 😅  
Basically what I was thinking was that it might confuse users to see the step  
From the drop-down menu, select **Use Docker Debug**. If the **Debug** tab
  isn't visible, select the **Exec** tab and then enable the **Debug mode**
  setting. 
Since if they didn't meet the requirements to use Docker Debug, they won't see the **Use Docker Debug** in the drop-down menu. 
 Again full apologies for such long comments. Please ignore if none of this made sense btw!

Reply:
Same as the other comment. I also rearranged it so that the sign in information appears in the steps rather than the intro paragraph.